### PR TITLE
Aggregate InstanceData in InstanceInfo and cleanup of IConfiguration.

### PR DESCRIPTION
**Pull request by arunagrawal84**
Move AWS specific properties to AwsInstanceInfo.java.

### CODE REVIEW

File: priam/src/main/java/com/netflix/priam/identity/config/InstanceInfo.java [hunk 1]
```diff
@@ -75,27 +70,31 @@
...
+     * Classic is deprecated environment by AWS.
      *
-     * @return the availability zone of the running instance. e.g. us-east-1c
-     * @throws JSONException
+     * @return Environment of the current running instance.
      */
-    String getAvailabilityZone() throws JSONException;
+    InstanceEnvironment getInstanceEnvironment();
+
+    enum InstanceEnvironment {
```
Comment (vinaykumarchella) [inline priam/src/main/java/com/netflix/priam/identity/config/InstanceInfo.java:96]: Having this in an interface might not make sense for non AWS environments.

File: priam/src/main/java/com/netflix/priam/identity/config/AWSInstanceInfo.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+    @Inject
+    public AWSInstanceInfo(ICredential credential) {
+        this.credential = credential;
+    }
+
+    @Override
+    public String getPrivateIP() {
+        if (privateIp == null)
+            privateIp =
```
Comment (vinaykumarchella) [inline priam/src/main/java/com/netflix/priam/identity/config/AWSInstanceInfo.java:57]: nit: missing `{` after if

File: priam/src/main/java/com/netflix/priam/identity/config/AWSInstanceInfo.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        if (vpcId == null)
+            try {
+                vpcId =
+                        SystemUtils.getDataFromUrl(
+                                        "http://169.254.169.254/latest/meta-data/network/interfaces/macs/"
+                                                + nacId
+                                                + "vpc-id")
+                                .trim();
+            } catch (Exception e) {
+                logger.info(
```
Comment (vinaykumarchella) [inline priam/src/main/java/com/netflix/priam/identity/config/AWSInstanceInfo.java:149]: Is it a `INFO` or `ERROR`

File: priam/src/main/java/com/netflix/priam/config/IConfiguration.java [hunk 4]
```diff
@@ -369,35 +371,37 @@ default int getRestoreThreads() {
     String getACLGroupName();
 
     /** @return true if incremental backups are enabled */
-    boolean isIncrBackup();
-
-    /** @return Get host IP */
-    String getHostIP();
+    default boolean isIncrBackup() {
```
Comment (vinaykumarchella) [inline priam/src/main/java/com/netflix/priam/config/IConfiguration.java:374]: might be more meaningful to have `isIncreBackupEnabled`?

File: priam/src/main/java/com/netflix/priam/PriamServer.java [hunk 5]
```diff
@@ -194,7 +196,7 @@ private void setUpSnapshotService() throws Exception {
     }
 
     public InstanceIdentity getId() {
```
Comment (sumanth-pasupuleti) [inline priam/src/main/java/com/netflix/priam/PriamServer.java:198]: nit: rename method as well to getInstanceIdentity?

File: priam/src/main/java/com/netflix/priam/config/IConfiguration.java [hunk 6]
```diff
@@ -762,7 +773,9 @@ default boolean useSudo() {
      *
      * @return SNS Topic ARN to be used to send notification.
      */
-    String getBackupNotificationTopicArn();
+    default String getBackupNotificationTopicArn() {
+        return "";
```
Comment (vinaykumarchella) [inline priam/src/main/java/com/netflix/priam/config/IConfiguration.java:777]: String.EMPTY?

**Review by vinaykumarchella — COMMENTED** [review]
minor nits

File: priam/src/main/java/com/netflix/priam/identity/config/InstanceInfo.java [hunk 7]
```diff
@@ -75,27 +70,31 @@
     String getVpcId(); // the id of the vpc for running instance
 
     /**
-     * AWS Account ID of running instance.
+     * Get the region of the AWS account of running instance
```
Comment (sumanth-pasupuleti) [inline priam/src/main/java/com/netflix/priam/identity/config/InstanceInfo.java:73]: remove AWS specific code/comments from this interface

**Review by sumanth-pasupuleti — COMMENTED** [review]
fix and ship

**Pull request merged**